internal/checker/checker.go

```diff
@@ -724,15 +724,21 @@ func checkPack(ctx context.Context, r restic.Repository, id restic.ID) error {
 			continue
 		}
 
-		n, err := r.Key().Decrypt(buf, buf)
+		if len(buf) < r.Key().NonceSize()+r.Key().Overhead() {
+			errs = append(errs, errors.Errorf("blob %v: invalid encrypted data, too short", i))
+			continue
+		}
+
+		nonceSize := r.Key().NonceSize()
+		nonce, ciphertext := buf[:nonceSize], buf[nonceSize:]
+		plaintext, err := r.Key().Open(ciphertext, nonce, ciphertext, nil)
```

Comment:
Also, since it’s an append API it should not overwrite the slice, but add data after it. That’s why you use ciphertext[:0].

Reply:
Ok, need to make that sure. Thanks!